# Introduction

Cancer genomics has entered a new era with the proliferation of vast amounts of open-source data, providing an unprecedented breadth of information. These datasets, ranging from gene expression profiles to proteomic signatures, hold immense potential for advancing our understanding of the molecular underpinnings of cancer and pinpointing novel research targets. 

The challenge, however, lies in harmonizing these diverse datasets to extract meaningful insights. The heterogeneity across datasets, stemming from variations in experimental platforms, data formats and modalities, and even biological contexts, poses a significant obstacle. As we delve into the depths of open-source repositories such as TCGA and others, the realization dawns that while each dataset is a valuable piece of the puzzle, the ability to integrate cross dataset surverys cohesively is key. 

Bridging the gaps between disparate datasets is not merely a technical hurdle but a fundamental necessity to harness the true power of open-source data for advancing cancer research. In this research endeavor, I aim to explore the breadth of open-source data in cancer genomics and address the inherent challenges associated with their integration. 

# Problem Statement

Despite the vast amount of open-source data available in cancer genomics, the integration of diverse datasets with cross dataset surverys remains a critical challenge. The heterogeneous nature of these datasets, stemming from variations in experimental platforms, data formats, and biological contexts, impedes the seamless synthesis of information. This fragmentation hinders comprehensive analyses that could unveil novel insights into the molecular mechanisms underlying cancer progression. Consequently, there is an urgent need for a approaches that integrat this diverse open-source data, enabling researchers to extract meaningful and actionable knowledge from these synthesized cancer genomics sources.

More specifically, the following research questions need to be addressed:
1. What open-source cancer genomics datasets are publicly available across various species, and what are the major barriers to their integration?
1. What challenges exist for cross dataset surveys and leveraging multiple dataset and modality of data?
1. How can these barriers be addressed to effectivley query accross multiple heterogenous data sources? 

# Objectives

1. Develop a generic and widely applicable tool for surveying open-source gene expression and proteomic data, allowing for exploration of expressions across diverse tissues and diseases in both human and animal models. 

1. The focus will be on creating a vignette that demonstrates how to answer a hypothetical question, specifically centered around gene X in the context of disease Y, across human and animal models. This vignette will be manifested as a body of code, serving as a programmatic output of the research. 

1. By deliberately keeping the scientific hypothesis less critical, the emphasis will shift towards the broader survey of open-source datasets, tooling, and the identification of limitations and challenges associated with cross-dataset work. The ultimate goal is to contribute a versatile and valuable resource to the field of bioinformatics, with the potential for widespread utility beyond any specific organizational context.

1. The creation of a deliverable to showcase my bioinformatics knowledge and skills. By utilizing Quarto as an open-source technical publishing system, I can demonstrate effective communication of project details, methods, and results through the website.

# Preliminary Literature Review

# Methodology

# References

https://medium.com/porism/whats-needed-to-discover-link-and-combine-open-datasets-a11890c4b0a0



