# A comparative evaluation of data-merging and meta-analysis methods for reconstructing gene-gene interactions

Lagani, V., Karozou, A.D., Gomez-Cabrero, D. et al. A comparative evaluation of data-merging and meta-analysis methods for reconstructing gene-gene interactions. BMC Bioinformatics 17 (Suppl 5), S194 (2016). https://doi.org/10.1186/s12859-016-1038-1

- Main takeaway: Combining data from heterogeneous data sources provides greater statistical power and insight than solely analyzing individual sets, however systemic biases need to be addressed to combine the sets. Both meta-analysis and data merging methods appear to be suitable for addressing systematic variations in studies that share the same experimental design. In this study both approaches outperformed (a) the na√Øve solution of merging data together ignoring possible biases, and (b) the results that are expected when only one dataset out of the available ones is analyzed in isolation. All studies in M follow the same experimental protocol, analyze the same type of biological specimens, and measure the same n expression values (probesets).

- In MA statistical methods are separately applied on each dataset for obtaining statistics of interest, e.g., differential expression p-values. The results from each study are then combined for creating summary statistics. The DM approach merges samples from different studies in a unique dataset, on which subsequent analyses are performed. While MA methods implicitly take in account batch-effects, DM require suitable Batch-Effect Removal (BER) algorithms

- Limitations: Firstly, Within-study batch- effects were only partially addressed, by pre-processing each single dataset with RMA. While the Quantile Normalization step included in the RMA algorithm should have removed at least part of the within-study biases, it is known that this approach is not optimal. Secondly, the design of the comparison slightly advantages DM method, particularly because all datasets belong to the same data collection and thus measure the same probesets. When this is not the case (e.g., when data from different microarray platforms are co-analyzed, DM method are not easily applicable, while MA methods can be straightforwardly used. 

- The authors  present a first, contrived case-study with partial reconstruction of the regulatory network of the Ikaros transcription factor from human PBMC data. This case-study presented several characteristics that made it harder to solve than the reconstruction of the E. coli and Yeast regulatory networks: **different cell-type proportions across subjects, a many-to- many correspondence between genes and probesets, the list of known interactions was partially derived from animal models instead than human data**. The authors used a constraint-based network reconstruction algorithm instead of relevance networks. Despite all these differences both Combat and Fixed-Effects method demonstrated to be able to retrieve subsets of genes significantly enriched for known Ikaros interactions and to outperform both the No-Correction and Single-Dataset approach.

- These findings are valuable to illustrate challenges in cross-surverying multiple human data sets and potential tactics to mitigate systematic variation in these scenarios. 