# Comparison of Merging and Meta-Analysis as Alternative Approaches for Integrative Gene Expression Analysis 
https://www.hindawi.com/journals/isrn/2014/345106/

Taminau, J., Lazar, C., Meganck, S., & Now√©, A. (2014). Comparison of merging and meta-analysis as alternative approaches for integrative gene expression analysis. International Scholarly Research Notices, 2014.

- Two different approaches for conducting large-scale analysis of microarray gene expression data‚Äîmeta-analysis and data merging‚Äîare compared in the context of the identification of cancer-related biomarkers, by analyzing six independent lung cancer studies. Within this study, the authors investigate the hypothesis that analyzing large cohorts of samples resulting in merging independent data sets designed to study the same biological problem results in lower false discovery rates than analyzing the same data sets within a more conservative meta-analysis approach. 
- The main difference between meta-analysis and the integrative analysis via data merging is visible in the next two steps: according to the meta-analysis approach each data set is analyzed individually and the final results are combined, while accord- ing to the merging approach the data sets are first combined in a unique, much larger data set and then analyzed. 
- integrative analysis means combining the information of multiple and independent studies, designed to study the same biological problem, in order to extract more general and more reliable conclusions. 

https://static.hindawi.com/articles/isrn/volume-2014/345106/figures/345106.fig.001a.jpg

- Issues for integrative analysis- many related to the retrieval and preprocessing of data
- For the merging approach- Before combining the expression values of different studies, they have to be made comparable to each other. Since the use of different experimentation plans, platforms, and methodologies by different research groups introduces undesired batch effects in the gene expression values an additional transformation of the data to remove those effects is needed. 
- the authors investigate both approaches in the context of the identification of lung cancer-related differentially expressed genes (DEGs) or biomarkers, that is, genes that have a discriminating profile in normal tissues versus lung cancer tissues. Those genes can provide new insights into the biological mechanisms of lung cancer and of cancer in general and can lead to promising biomarkers and new directions for drug development or treatments. The authors select a number of already published and publicly available data sets containing both normal and cancer tissues and identify robust and stable DEGs using both approaches for integrative analysis. 
- Use of InSilico DB
- A list of potential data sets was programmatically retrieved from the InSilico DB using the getDatasetList function from the R/Bioconductor inSilicoDb package
- 2 different platforms : Only studies assayed on Affymetrix Human Genome U133A (GPL96) and Affymetrix Human Genome U133 Plus 2.0 (GPL570) were taken into consideration. 
- In order to ensure the robustness of the found DEG lists and to encounter false-positive discoveries the authors implemented an extra resampling step on top of the limma method. In each iteration, they arbitrarily keep 90% of the samples and apply limma to obtain a DEG list fulfilling the two above- mentioned criteria. After ùëõ resampling iterations they obtained ùëõ different DEG lists and the final DEG list will be the intersection of those lists. Taking the intersection might seem rather strict but they empirically confirmed a convergence of the number of DEGs after around 50 resampling iterations, depending on the quality of the study. A resampling iteration size ùëõ of 100 was used for all experiments. 
- For the merging approach, the authors first merge all six studies into one global data set using the following batch effect removal methods: NONE (no batch effect removal), BMC (batch-mean centering), COM- BAT (empirical bayes), DWD (distance-weighted dis- crimination), and XPN (cross-platform normalization. Then applied to each merged data set the same procedure to find robust DEGs. 
- Use of resampling as extra validation effect: The rationale behind this extra validation step for the biomarkers discovery is the fact that a stable biomarker should be identified even by making small perturbations in the data set, caused by removing systematically random samples. 
- Addresses how higher # of DEGs found for one platform versus another - speaks to scenarios that can arise when using different platforms for MA approach
- for Merging approach: Within this study applied six different batch effect removal methods resulting in six different lists of DEGs. , the applied same resampling procedure
- difference in number of DEGs with and without resampling is less prominent than in the meta-analysis approac 
- Except for XPN method, methods of BMC, COMBAT, and DWD are not able to find more DEGs than when no batch effect removal at all is performed. 
- MDS scaling plot does show clearly apparent batch affects - technical based batch effects
- it could be expected that this undesirable effect would influence the discovery of DEGs. Based on the results this is however not the case and the explanation lies in the fact that MDS plots provide a global view on the data, while the identification of DEGs is more based on local effects, that is, the specific expression of one gene in certain conditions. The explanation for this apparently paradoxical result is in the fact that not all genes are affected by batch effect removal in the same way. Genes which are differentially expressed in individual studies could still remain differentially expressed after data merging with no batch effect correction, if their expression is not much affected by batch effects. Moreover, even if a gene is affected by batch effects the authors observed that the difference between the two modes or conditions of the gene (in the case control versus lung cancer) is almost always preserved over all samples of the merged data set. 
- Showcase a DEG that is apparent BEFORE batch effect removal and disappears after (present in 3 samples from 1 microarray platform) - one can argue that this signal is not consistent across all individual studies and can be due to a technical, platform- dependent artifact. 

- Showcase a DEG that is not apparent UNTIL batch effect removal - instead of a technical artifact, it is actually the batch effect that distorts the global expression profile of the LRRN3 gene. 

# Conclusion:
## Both meta-analysis and merging approaches are able to find differentially expressed genes (DEGs) consistently expressed in all individual data sets. In both approaches a resampling or bootstrapping framework is needed to avoid false positives and to ensure robust gene lists. 
- Although batch effects were clearly present when merging the different data sets, they were not hindering the identification of DEGs. This surprising finding is however not generalizable to the clustering or classification of tasks since it is very depending on the specific application. 
- The authors conclude that significantly more DEGs are identified through merging. Moreover, all 25 identified DEGs through meta-analysis are also identified in the merging approach. Most genes in both lists were scanned in the literature and showed to play a role or at least be involved in the development of lung cancer and can be further validated and used in clinical applications. 