# Recent developments and future directions in meta-analysis of differential gene expression in livestock RNA-Seq
Keel, B. N., & Lindholm-Perry, A. K. (2022). Recent developments and future directions in meta-analysis of differential gene expression in livestock RNA-Seq. Frontiers in Genetics, 13, 983043.

https://www.frontiersin.org/articles/10.3389/fgene.2022.983043/full

- inter- study variability due to technical differences (e.g., sample preparation, library protocols, batch effects) as well as biological differences (e.g., environmental, management, and genetic effects) also contributes to reproducibility issues. 

- The first two issues are related to data acquisition and pre-processing, while the remaining issues relate to choosing a meta-analysis procedure and interpretation of results. 

## Key steps and important considerations in the RNA-Seq meta-analysis workflow 
1. Identifying data sets and data acquisition - One major issue with data in publicly available omics repositories, which is not unique to livestock, is the incompleteness of dataset descriptions, called metadata. Metadata can include many different pieces of information, including sources of data, dates of data collection, methods used, etc. 
2. Data preprocessing - there is no optimal pipeline for the variety of different applications and analysis scenarios in which RNA-Seq can be used, and preprocessing protocols may vary greatly from study to study. For meta-analysis, data preprocessing should be as standardized as possible between studies in order to minimize technical heterogeneity. Although preprocessed data is often available in public repositories, it is best to start from the raw data and process all data sets using a unified pipeline. Standard quality control, read mapping, and quantification procedures should be performed on raw RNA-Seq data at the start of any meta-analysis (Conesa et al., 2016). Briefly, raw reads should be analyzed for sequence quality, GC content, adaptor presence, overrepresented reads, and duplicated reads. Acceptable levels of duplicated content differ by organism but should be homogeneous for samples in the same experiment. It is recommended that outliers with more than 30% disagreement be discarded (Conesa et al., 2016). After initial quality checking of raw sequence reads, reads should be trimmed to remove adaptor sequences, low-quality reads, and poor-quality bases. Trimmed reads should then be mapped to the reference genome, and libraries with low read mapping percentage (organism dependent) removed from downstream analysis. After read mapping, quantification of transcript/gene expression should be performed. 

- how to handle data generated using different RNA library preparation protocols ? 
- rRNA depleted libraries systematically produced lower estimates of the relative expression of protein-coding genes. Using a common processing pipeline, in particular a common threshold for filtering lowly expressed genes, for data sets produced using differing RNA selection methods would then result in incompatible downstream data. However, equivalent expression levels between polyA-selected and rRNA-depleted libraries can be achieved using a combination of reference transcriptome filtering and a ratio-based correction. 

3. Choosing a meta-analysis procedure 
- For microarray data, proposed meta-analysis methods include p-value combination (Marot et al., 2009), combining effect sizes (Choi et al., 2003), and ranking genes within each study 
- Commonly used meta-analysis procedures for RNA-Seq data include p-value combination and generalized linear models (GLM) with a fixed study effect. 
- use of p-value combination is likely the best choice for most meta- analyses (given results from Melanoma study)
- Additional data filtering is required if using the p-value combination approach. A critical underlying assumption for the statistics in p-value combination is that p-values for all genes in the per-study differential analyses are uniformly distributed under the null hypothesis (Rau et al., 2014). This assumption is, however, not always satisfied for RNA-Seq data; in particular, a peak is often observed for p-values close to one due to the discretization of p-values for very low counts (Rau et al., 2013). To circumvent this issue, weakly expressed genes should be filtered from the analysis. Data-based methods, such as HTSFilter (Rau et al., 2013), are preferred over ad-hoc filtering procedures, as they can account for differences between studies resulting from sequencing depth, intra- condition variability, and other technical factors. 


4. Interpreting meta-analysis results - 
    1. Fold changes - not uncommon for genes in a meta-analysis to exhibit conflicting expression patterns among studies. Fold changes must be handled differently for global GLM and p-value combination techniques. fold changes for the p-value combination are calculated on a per-study basis and can exhibit differing patterns between studies. In the context of microarrays, approaches for overcoming conflicting expression patterns between studies for the inverse normal (Marot et al., 2009) and Fisher (Owen, 2009) p-value combination methods have been proposed. Both methods rely on the use of a two-tailed gene expression distribution, where under- and over-expressed genes reside in the tails of the distribution. However, for RNA-Seq data, where the data follows an overdispersed Poisson or negative binomial distribution, under- and over-expressed genes cannot be separated into distribution tails. It has been suggested that DEG with differing expression patterns between studies be removed post hoc (Rau et al., 2014). However, genes with both concordant and discordant gene expression patterns across studies could be of interest. For example, sample collection time points will often be different between studies. The expression of some genes may be higher at earlier sampling time points and lower at later sampling time points. 
    2. Robustness of differentially expressed genes in p-value combination - Differential expression results, especially those derived from small sample sizes, are known to be susceptible to heterogeneity (Cui et al., 2021). As a result, reproducibility of DEG is often poor from study to study. For this reason, robustness of DEG arising from meta-analysis should be measured by replication validity rather than in independent data, such as cross-validation. Jackknife sensitivity analysis consists of repeating the meta- analysis procedure multiple times, each time removing a single study from the analysis - A gene that fails only one jackknife analysis indicates that the meta p-value was being driven by the p-value of a single study, i.e., there is a significant amount of study bias. Similarly, genes that fail multiple jackknife analyses are being driven by p-values from multiple studies. This indicates a reduced level of study bias, i.e. a more robust result.
    3. Downstream analysis - most popular type of downstream analysis is gene set enrichment analysis (GSEA) . Over-representation analysis (ORA) approaches statistically evaluate the fraction of genes in a particular ontology/pathway found among the set of DEG. The most commonly used tests in ORA are based on the hypergeometric, chi-square, or binomial distribution (Huang et al., 2009). ORA methodologies only utilize lists of DEG, disregarding other quantitative measures of the genes such as fold change. In functional class scoring (FCS) methods, use a similar approach to ORA but adds in quantitative information from the genes (Mootha et al., 2003). A gene score is calculated for each gene, and individual gene scores are used to calculate a gene set score. Significance of gene set scores is assessed and differentially enriched gene sets are reported. Topology-based (TB) methods for GSEA go a step further by utilizing the network structure of a gene pathway to quantify a geneâ€™s importance to a given pathway (Draghici et al., 2007). Currently, researchers are using the same downstream analyses for DEG sets coming from meta-analysis as are used for single studies. While ORA tools are directly applicable to meta-analysis results, current FCS and TB methodologies are not since they are designed to be used with a single set of information for each gene, e.g., a single fold change. Integration of DEG expression profiles across studies in a meta-analysis could shed light on the molecular mechanisms governing phenotypes across environments, especially for DEG exhibiting discordant fold changes across studies. To our knowledge, there are currently no bioinformatic approaches that integrate meta-analysis results in this way. Future development of algorithms to incorporate this type of data would help harness the full potential of gene expression meta-analyses. 