# Potential Data Sources

## The Cancer Genome Atlas Program (TCGA)
- https://www.cancer.gov/ccg/research/genome-sequencing/tcga
- Publicly available genomic, epigenomic, transcriptomic, and proteomic data from molecularly characterization of over 20,000 primary cancer and matched normal samples spanning 33 cancer types  
- Harmonized Cancer Datasets Genomic Data Commons Data Portal: https://portal.gdc.cancer.gov

## NIH Proteomic Data Commons
- https://datacommons.cancer.gov/repository/proteomic-data-commons

## Cancer Data Service (CDS) 
- One of several data commons within the Cancer Research Data Commons (CRDC).
- The CDS provides data storage and sharing capabilities for NCI-funded studies that fall under the following categories: Studies with data that do not match an existing CRDC data commons and studies with data that do not fit current data type criteria and/or the minimum metadata standards for a CRDC data commons.
- ex: Childhood Cancer Data Initiative (CCDI) as well as data from independent research projects. 

## Gene Expression Omnibus (GEO)
- Gene Expression Omnibus is a database for gene expression profiling and RNA methylation profiling managed by the National Center for Biotechnology Information. These high-throughput screening genomics data are derived from microarray or RNA-Seq experimental data.
- https://www.ncbi.nlm.nih.gov/geo/
- Can choose organism as homo sapiens
- Has data sets from conducted studies, for example: gene dysregulation induced by the exposure of MCF-7 breast cancer cells to hyperglycemia, or a combination of hyperglycemia and hyperinsulinemia was analyzed using Microarray gene expression assay. https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE179768

## Expression Atlas
- https://www.ebi.ac.uk/gxa/home
- Expression Atlas is an added value database that provides information about gene and protein expression in different species and contexts, such as tissue, developmental stage, disease or cell type. 
- Expression Atlas uses microarray and RNA-Seq data sets from ArrayExpress, NCBIâ€™s Gene Expression Omnibus (GEO), European Nucleotide Archive (ENA) as well as other sources.

## Gemma
- https://gemma.msl.ubc.ca/home.html
- Gemma is an open source database, analysis software system and web site for expression data re-use and meta-analysis. It currently contains analyzed data from over 3300 expression profiling studies, yielding billions of data points that can be searched and explored. Gemma offers web-based access to differential expression results and coexpression data in a meta-analysis framework.

## re3data.org
- A searchable database of more than 2,000 research data repositories, including 100+ relating to cancer.

## Genotype-Tissue Expression Program (GTEx)
- https://www.gtexportal.org/home/ 
- For non-diseased datasets

## Google Dataset Search
https://datasetsearch.research.google.com

Some of these potential data sources were listed at https://libguides.mskcc.org/artificial_intelligence/datasets, which has additional data search engines and sets that could be of use. 

Gemma and re3data need to be verified for legitamacy. 